File: C++/DM8/src/graphics/layers/Group.cpp
#include "Group.h"


namespace DM8 {
	namespace graphics {
			
		Group::Group(const math::Mat4& transform)
			:m_TransformationMatrix(transform)
		{

		}

		Group::~Group()
		{
			for (int i = 0; i <= m_Renderables.size(); i++)
				delete m_Renderables[i];
		}

		void Group::Push(Renderable2D* renderable)
		{
			m_Renderables.push_back(renderable);
		}

		void Group::Submit(Renderer2D* renderer)
		{
			renderer->Push(m_TransformationMatrix);
			for (Renderable2D* renderable : m_Renderables) {
				renderable->Submit(renderer);
			}
			renderer->Pop();
		}

	}
}

File: Java/src/net/mirchs/dm8/graphics/layers/Group.java
package net.mirchs.dm8.graphics.layers;

import java.util.ArrayList;
import java.util.List;

import net.mirchs.dm8.graphics.Renderable2D;
import net.mirchs.dm8.graphics.Renderer2D;
import net.mirchs.dm8.math.Matrix4f;

public class Group extends Renderable2D {

	private List<Renderable2D> renderables;
	private Matrix4f transformationMatrix;
	
	public Group(Matrix4f transform) {
		renderables = new ArrayList<Renderable2D>();
		this.transformationMatrix = transform;
	}
	
	public void add(Renderable2D renderable) {
		renderables.add(renderable);
	}

	public void submit(Renderer2D renderer) {
		renderer.push(transformationMatrix, false);
		for (Renderable2D renderable : renderables) {
			renderable.submit(renderer);
		}
		renderer.pop();
	}

}
